crypto/x509/by_file.c

```diff
@@ -121,34 +118,37 @@ int X509_load_cert_file_ex(X509_LOOKUP *ctx, const char *file, int type,
                     break;
                 } else {
                     ERR_clear_last_mark();
+                    if (count == 0)
+                        ERR_raise(ERR_LIB_X509, X509_R_NO_CERTIFICATE_FOUND);
                     goto err;
                 }
             }
             ERR_clear_last_mark();
-            i = X509_STORE_add_cert(ctx->store_ctx, x);
-            if (!i)
+            if (!X509_STORE_add_cert(ctx->store_ctx, x)) {
+                if (count == 0)
+                    ERR_raise(ERR_LIB_X509, X509_R_NO_CERTIFICATE_FOUND);
```

Comment:
IMO this is wrong and misleading to report this error reason here.

Reply:
For example, we successfully added 1 certificate using `X509_STORE_add_cert()`, but when adding a second certificate, `X509_STORE_add_cert()` returned an error (return code 0). What should the `X509_load_cert_file_ex()` return? 1 (the number of correctly added certificates) or 0 (the same error occurred)? It seems that the function should return 1, but I would like to know your opinion?

Reply:
I think the return value should still be 0 - failing X509_STORE_add_cert() indicates some fatal problem such as allocation failure or memory corruption. It should not happen in any normal circumstances. So forcing count = 0 would be appropriate.